app/code/Magento/PaypalGraphQl/Model/Plugin/Cart/PayflowPro/SetPaymentMethodOnCart.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\PaypalGraphQl\Model\Plugin\Cart\PayflowPro;
+
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Query\Resolver\ContextInterface;
+use Magento\Paypal\Model\Config;
+use Magento\Quote\Model\Quote;
+use Magento\QuoteGraphQl\Model\Cart\Payment\AdditionalDataProviderPool;
+use Magento\Sales\Model\Order\Payment\Repository as PaymentRepository;
+use Magento\PaypalGraphQl\Observer\PayflowProSetCcData;
+
+/**
+ * Set additionalInformation on payment for PayflowPro method
+ */
+class SetPaymentMethodOnCart
+{
+    /**
+     * @var PaymentRepository
+     */
+    private $paymentRepository;
+
+    /**
+     * @var AdditionalDataProviderPool
+     */
+    private $additionalDataProviderPool;
+
+    /**
+     * @param PaymentRepository $paymentRepository
+     * @param AdditionalDataProviderPool $additionalDataProviderPool
+     */
+    public function __construct(
+        PaymentRepository $paymentRepository,
+        AdditionalDataProviderPool $additionalDataProviderPool
+    ) {
+        $this->paymentRepository = $paymentRepository;
+        $this->additionalDataProviderPool = $additionalDataProviderPool;
+    }
+
+    /**
+     * Set redirect URL paths on payment additionalInformation
+     *
+     * @param \Magento\QuoteGraphQl\Model\Cart\SetPaymentMethodOnCart $subject
+     * @param mixed $result
+     * @param Quote $cart
+     * @param array $paymentData
+     * @param ContextInterface $context
+     * @return void
+     * @throws GraphQlInputException
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function afterExecute(
+        \Magento\QuoteGraphQl\Model\Cart\SetPaymentMethodOnCart $subject,
+        $result,
+        Quote $cart,
+        array $paymentData,
+        ContextInterface $context
+    ): void {
+        $paymentData = $this->additionalDataProviderPool->getData(Config::METHOD_PAYFLOWPRO, $paymentData);
+
+        if (!$context->getExtensionAttributes()->getIsCustomer()
```

Comment:
Hi @Usik2203 . I see you are changing the message signatures to pass in context with the objective to check if cart belongs to a customer. Changing method signatures might also impact other plugins. Try limiting context to resolver level and using it in a plugin is probably not ideal. To get around this, $cart should have the customer related metadata
```
$cartCustomerId = (int)$cart->getCustomerId();

        /* Guest cart, allow operations */
        if (0 === $cartCustomerId && (null === $customerId || 0 === $customerId)) {
            return $cart;
        }
```
You can use the above.

Reply:
Hi @prabhuram93  
Yes , you are right , I can get check customer is logged or not using `$cart` 
Thank you for review and advice.

Reply:
No worries @Usik2203. Code looks neat now. Good work! Lets get the builds green.